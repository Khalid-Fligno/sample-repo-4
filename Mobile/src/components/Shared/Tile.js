import React from "react";
import {
  View,
  ImageBackground,
  Text,
  TouchableOpacity,
  StyleSheet,
  Dimensions,
  Animated,
} from "react-native";
import PropTypes from "prop-types";
import colors from "../../styles/colors";
import fonts from "../../styles/fonts";

const { width } = Dimensions.get("window");
var tileHeight = 200;

const Tile = (props) => {
  const {
    onPress,
    title1,
    image,
    disabled,
    showTitle,
    overlayTitle,
    height,
    customContainerStyle,
    showTitleStyle,
    overlayTitleStyle,
    imageUrl,
  } = props;

  const animatedValue = new Animated.Value(1);

  const animatedStyle = {
    transform: [{ scale: animatedValue }],
  };

  const handlePressIn = () => {
    Animated.spring(animatedValue, {
      toValue: 0.92,
      useNativeDriver: true,
    }).start();
  };
  const handlePressOut = () => {
    Animated.spring(animatedValue, {
      toValue: 1,
      friction: 3,
      tension: 40,
      useNativeDriver: true,
    }).start();
  };

  return (
    <TouchableOpacity
      disabled={disabled}
      onPress={onPress}
      style={[styles.cardContainer, customContainerStyle]}
      onPressIn={handlePressIn}
      onPressOut={handlePressOut}
    >
      <Animated.View style={[styles.flexContainer, animatedStyle]}>
        <ImageBackground
          source={imageUrl ? { uri: imageUrl, cache: "force-cache" } : image}
          style={styles.image}
        >
          {overlayTitle && (
            <View style={styles.opacityLayer}>
              <Text style={[styles.title, overlayTitleStyle]}>
                {title1.toUpperCase()}
              </Text>
            </View>
          )}
        </ImageBackground>
      </Animated.View>
      <View>
        {showTitle && (
          <Text style={[styles.title, showTitleStyle]}>{title1}</Text>
        )}
      </View>
    </TouchableOpacity>
  );
};

export default Tile;

Tile.propTypes = {
  onPress: PropTypes.func.isRequired,
  title1: PropTypes.string.isRequired,
  image: PropTypes.number.isRequired,
  disabled: PropTypes.bool,
  height: PropTypes.number,
  customContainerStyle: PropTypes.object,
  showTitleStyle: PropTypes.object,
  overlayTitleStyle: PropTypes.object,
  imageUrl: PropTypes.string,
};

Tile.defaultProps = {
  disabled: false,
};

const styles = StyleSheet.create({
  cardContainer: {
    flex: 0.25,
    margin: 5,
    marginTop: 0,
    height: tileHeight,
  },
  flexContainer: {
    flex: 1,
  },
  image: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
    borderRadius: 2,
    overflow: "hidden",
  },
  opacityLayer: {
    flex: 1,
    width: "100%",
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: colors.transparentBlackLightest,
  },
  title: {
    fontFamily: fonts.bold,
    fontSize: 20,
    color: colors.black,
    marginTop: 10,
    marginBottom: 10,
    width: "100%",
    textTransform: "capitalize",
  },
});
