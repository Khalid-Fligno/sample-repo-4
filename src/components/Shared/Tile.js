import React from 'react';
import {
  View,
  ImageBackground,
  Text,
  TouchableOpacity,
  StyleSheet,
  Dimensions,
  Animated,
} from 'react-native';
import PropTypes from 'prop-types';
import colors from '../../styles/colors';
import fonts from '../../styles/fonts';

const { width } = Dimensions.get('window');
var tileHeight = 200;
export default class Tile extends React.PureComponent {
  constructor(props) {
    super(props);
    this.animatedValue = new Animated.Value(1);
  }
  handlePressIn = () => {
    Animated.spring(this.animatedValue, {
      toValue: 0.92,
      useNativeDriver: true 
    }).start();
  }
  handlePressOut = () => {
    Animated.spring(this.animatedValue, {
      toValue: 1,
      friction: 3,
      tension: 40,
      useNativeDriver: true 
    }).start();
  }
  render() {
    const {
      onPress,
      title1,
      image,
      disabled,
      showTitle,
      overlayTitle,
      height,
      customContainerStyle,
      showTitleStyle,
      overlayTitleStyle,
      imageUrl
    } = this.props;
    tileHeight = height
    const animatedStyle = {
      transform: [{ scale: this.animatedValue }],
    };
    return (
      <TouchableOpacity
        disabled={disabled}
        onPress={onPress}
        style={[styles.cardContainer,customContainerStyle]}
        onPressIn={this.handlePressIn}
        onPressOut={this.handlePressOut}
      >
        <Animated.View
          style={[styles.flexContainer, animatedStyle]}
        >
          <ImageBackground
            source={imageUrl?{uri:imageUrl}:image}
            style={styles.image}
          >
            {
              overlayTitle && (<View style={styles.opacityLayer}>
                <Text style={[styles.title,overlayTitleStyle]}>
                 {title1.toUpperCase()}
                </Text>
              </View>)
            } 
          </ImageBackground>
        </Animated.View>
        <View>
         { showTitle && (<Text style={[styles.title,showTitleStyle]}>
            {title1}
          </Text>)}
        </View>
      </TouchableOpacity>
    );
  }
}

Tile.propTypes = {
  onPress: PropTypes.func.isRequired,
  title1: PropTypes.string.isRequired,
  image: PropTypes.number.isRequired,
  disabled: PropTypes.bool,
  height: PropTypes.number,
  customContainerStyle: PropTypes.object,
  showTitleStyle:PropTypes.object,
  overlayTitleStyle:PropTypes.object,
  imageUrl:PropTypes.string
};

Tile.defaultProps = {
  disabled: false,
};

const styles = StyleSheet.create({
  cardContainer: {
    flex: 0.25,
    margin: 5,
    marginTop:0,
    // width,
    shadowColor: colors.charcoal.standard,
    shadowOpacity: 0.5,
    shadowOffset: { width: 0, height: 2 },
    shadowRadius: 4,
    height:tileHeight
  },
  flexContainer: {
    flex: 1,
  },
  image: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 2,
    overflow: 'hidden',
  },
  opacityLayer: {
    flex: 1,
    width: '100%',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: colors.transparentBlackLightest,
  },
  title: {
    fontFamily: fonts.bold,
    fontSize: 20,
    color: colors.black,
    // shadowColor: colors.black,
    // shadowOpacity: 1,
    marginTop:10,
    marginBottom:10,
    // shadowOffset: { width: 0, height: 0 },
    // shadowRadius: 5,
    width:'100%',
    textTransform:'capitalize'
  },
});
