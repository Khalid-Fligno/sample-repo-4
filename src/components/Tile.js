import React from 'react';
import { View, ImageBackground, Text, TouchableOpacity, StyleSheet, Dimensions } from 'react-native';
import PropTypes from 'prop-types';
import colors from '../styles/colors';
import fonts from '../styles/fonts';

const { width } = Dimensions.get('window');

const Tile = ({
  onPress,
  title1,
  image,
}) => (
  <TouchableOpacity
    onPress={onPress}
    style={styles.cardContainer}
  >
    <ImageBackground
      source={image}
      style={styles.image}
    >
      <View style={styles.titleContainer}>
        <Text style={styles.title}>
          {title1.toUpperCase()}
        </Text>
      </View>
    </ImageBackground>
  </TouchableOpacity>
);

Tile.propTypes = {
  onPress: PropTypes.func.isRequired,
  title1: PropTypes.string.isRequired,
  image: PropTypes.number.isRequired,
};

const styles = StyleSheet.create({
  cardContainer: {
    flex: 1,
    margin: 5,
    paddingLeft: 10,
    paddingRight: 10,
    width,
    shadowColor: colors.charcoal.standard,
    shadowOpacity: 0.3,
    shadowOffset: { width: 0, height: 2 },
    shadowRadius: 4,
  },
  image: {
    flex: 1,
    alignItems: 'flex-start',
    justifyContent: 'center',
    paddingLeft: 25,
    borderRadius: 2,
    overflow: 'hidden',
  },
  titleContainer: {
    backgroundColor: 'rgba(255, 255, 255, 0.65 )',
    paddingTop: 8,
    paddingRight: 12,
    paddingBottom: 3,
    paddingLeft: 12,
    borderRadius: 2,
  },
  title: {
    fontFamily: fonts.bold,
    fontSize: 24,
    color: colors.black,
    textAlign: 'center',
  },
});

export default Tile;
