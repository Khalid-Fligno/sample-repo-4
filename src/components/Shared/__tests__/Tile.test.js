import React from 'react';
import renderer from 'react-test-renderer';
import Tile from '../Tile';

describe('Tile', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Tile
      onPress={jest.fn()}
      title1="Test"
      image={1}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
