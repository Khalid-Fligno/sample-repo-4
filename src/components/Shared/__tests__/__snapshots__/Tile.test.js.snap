// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 0.25,
      "margin": 5,
      "opacity": 1,
      "paddingLeft": 10,
      "paddingRight": 10,
      "shadowColor": "rgb(38, 38, 40)",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.5,
      "shadowRadius": 4,
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 2,
          "flex": 1,
          "justifyContent": "center",
          "overflow": "hidden",
        }
      }
    >
      <Image
        source={1}
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": undefined,
            },
            undefined,
          ]
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0, 0, 0, 0.5 )",
            "flex": 1,
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "rgb(255, 255, 255)",
              "fontFamily": "GothamBold",
              "fontSize": 24,
              "shadowColor": "rgb(0, 0, 0)",
              "shadowOffset": Object {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 5,
              "textAlign": "center",
            }
          }
        >
          TEST
        </Text>
      </View>
    </View>
  </View>
</View>
`;
